python/cudf/cudf/_libxx/copying.pyx

```diff
@@ -34,3 +144,464 @@ def gather(Table source_table, Column gather_map):
         column_names=source_table._column_names,
         index_names=source_table._index._column_names
     )
+
+
+def _scatter_table(Table source_table, Column scatter_map,
+                   Table target_table, bounds_check=True):
+
+    cdef table_view source_table_view = source_table.data_view()
+    cdef column_view scatter_map_view = scatter_map.view()
+    cdef table_view target_table_view = target_table.data_view()
+    cdef bool c_bounds_check = bounds_check
+
+    cdef unique_ptr[table] c_result
+
+    with nogil:
+        c_result = move(
+            cpp_copying.scatter(
+                source_table_view,
+                scatter_map_view,
+                target_table_view,
+                c_bounds_check
+            )
+        )
+
+    out_table = Table.from_unique_ptr(
+        move(c_result),
+        column_names=target_table._column_names,
+        index_names=None
+    )
+
+    out_table._index = (
+        None if target_table._index is None else target_table._index.copy(
+            deep=True)
+    )
+
+    return out_table
+
+
+def _scatter_scalar(scalars, Column scatter_map,
+                    Table target_table, bounds_check=True):
+
+    cdef vector[unique_ptr[scalar]] source_scalars
+    cdef bool c_bounds_check = bounds_check
+    cdef Scalar slr
+    for val, col in zip(scalars, target_table._columns):
+        slr = Scalar(val, col.dtype)
+        source_scalars.push_back(move(slr.c_value))
+    cdef column_view scatter_map_view = scatter_map.view()
+    cdef table_view target_table_view = target_table.data_view()
+
+    cdef unique_ptr[table] c_result
+
+    with nogil:
+        c_result = move(
+            cpp_copying.scatter(
+                source_scalars,
+                scatter_map_view,
+                target_table_view,
+                c_bounds_check
+            )
+        )
+
+    out_table = Table.from_unique_ptr(
+        move(c_result),
+        column_names=target_table._column_names,
+        index_names=None
```

Comment:
I am passing only data_view, so the returned column will not have index columns, and passing index_names will try to access starting columns as index which would cause problem. In the next line I am assigning the index from target to out_table to preserve the index.